Training/Training/Training.c

```diff
@@ -1,14 +1,138 @@
-// ------------------------------------------------------------------------------------------------
-// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
-#include<stdio.h>
-
-int main () {
-	printf ("Hello World!");
-	return 0;
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Training.c
+// Implements selection sort and binary search
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+
+#include"header.h"
+#include<stdbool.h>
+#include<stdio.h>
+#include<string.h>
+#include<errno.h>
+#include<malloc.h>
+#include<stdlib.h>
+
+/// <summary>Checks if a given input is a valid integer</summary>
+void IsValidInteger (char* prompt, int* num) {
+   bool output;
+   do {
+      char arrSizeStr[13], * result = NULL;
+      printf ("%s", prompt);
+      fgets (arrSizeStr, sizeof (arrSizeStr), stdin);
+      result = strchr (arrSizeStr, '\n');
+      if (result == NULL) {
+         while (getchar () != '\n');
+         output = false;
+      } else {
+         errno = 0;
+         char* endptr = NULL;
+         *num = strtol (arrSizeStr, &endptr, 10);
+         output = endptr == arrSizeStr || *endptr != '\n' || errno == ERANGE || arrSizeStr[0] == ' ' ||
+            arrSizeStr[0] == '\t' ? false : true;
+      }
+      if (output == false)
+         printf ("Invalid Input\n\n");
+   } while (output == false);
+}
+
+/// <summary>Accepts user inputs</summary>
+void UserInputs (int** arr, int* arrSize) {
+   printf ("\n");
+   do {
+      IsValidInteger ("Enter an array size: ", arrSize);

```

Comment:
Apply this wherever possible.
```suggestion
   do {
      IsValidInteger ("\nEnter an array size: ", arrSize);

```